chromadb/api/types.py

```diff
@@ -197,6 +211,37 @@ class EmbeddingFunction(Protocol[D]):
     def __call__(self, input: D) -> Embeddings:
         ...
 
+    def max_token_input_length(self) -> int:
+        return -1
+
+    def _check_large_inputs(
+        self: "EmbeddingFunction",
+        inputs: Documents,
+        token_count_function: Callable[..., Any] = None,
+    ) -> None:
+        if token_count_function is None:
+            return
+        if token_count_function is None:
+            return
+        if not isinstance(inputs, list):
+            return
+        # Check if all inputs are strings
+        if not all(isinstance(i, str) for i in inputs):
+            return
+
+        for i, t in enumerate(inputs):
+            tokens = token_count_function(t)
+            _too_long = (
+                len(tokens) > self.max_input_length()
+                if self.max_input_length() > 0
+                else False
+            )
+            if _too_long:
+                logger.warning(
+                    "WARNING: The following document exceed the maximum input size of "
+                    f"{self.max_input_length()}: id = {i}, text = {t[:50]}..."
```

Comment:
this `id` doesn't have anything to do with the `id` of the document in the collection - we should not refer to it that way.